libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigationProvider.kt

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.navigation.core
+
+import android.content.Context
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.navigation.base.options.NavigationOptions
+
+/**
+ * Singleton responsible for ensuring there is only one MapboxNavigation instance.
+ */
+object MapboxNavigationProvider {
+    @Volatile
+    private var mapboxNavigation: MapboxNavigation? = null
+
+    /**
+     * Create MapboxNavigation with provided options.
+     * Should be called before [navigation]
+     *
+     * @param context
+     * @param navigationOptions
+     * @param locationEngine
+     * @param locationEngineRequest
+     */
+    @JvmOverloads
+    @JvmStatic
+    fun createNavigation(
```

Comment:
Looking at this and how it'll be used, what about calling this `create` only? `MapboxNavigationProvider#create` reads pretty well. What do you think?